lib/widgets/store.dart

```diff
@@ -246,3 +249,36 @@ class LoadingPage extends StatelessWidget {
     return const Center(child: CircularProgressIndicator());
   }
 }
+
+/// A [State] that uses the ambient [PerAccountStore].
+///
+/// The ambient [PerAccountStore] can be replaced in some circumstances,
+/// such as when an event queue expires. See [PerAccountStoreWidget.of].
+/// When that happens, stateful widgets should
+/// - remove listeners on the old [PerAccountStore], and
+/// - add listeners on the new one.
+/// Use this mixin, overriding [onNewStore], to do that concisely.
+// TODO(#185): Until #185, I think [PerAccountStoreWidget.of] never actually
+//   returns a different [PerAccountStore] from one call to the next.
+//   But it will, and when it does, we want our [StatefulWidgets] to handle it.
+mixin PerAccountStoreAwareStateMixin<T extends StatefulWidget> on State<T> {
+  PerAccountStore? _store;
+
+  /// Called in the first [State.didChangeDependencies] call, and afterward
+  /// when [PerAccountStoreWidget.of] returns a different store than last time.
+  ///
+  /// In this, remove any listeners on the old store
+  /// and add them on the new store.
+  void onNewStore();
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final storeNow = PerAccountStoreWidget.of(context);
+    if (_store != storeNow) {
+      onNewStore();
+    }
+    _store = storeNow;
```

Comment:
It'd be good to have a test for this logic: make a test widget that uses this mixin, put it through its paces, check that `onNewStore` gets called at the appropriate times.

I think you can arrange such a test by moving the widget from under one PerAccountStoreWidget to under another with a different `accountId`. (Using `key` to ensure that the PerAccountStoreWidget's element gets replaced with a new one, rather than updated, so that its `initState` runs again; and also to ensure that the child element gets moved instead of replaced with a new one.) That isn't exactly something I anticipate we'll do in real life, but I think it's close enough to make an effective test.

Reply:
> That isn't exactly something I anticipate we'll do in real life

I might even go farther: couldn't we actually have the app code assert that we *don't* do that in real life? 😅

If we did ask it to do that, maybe accidentally, I think I'd wonder whether a piece of UI is showing data for the wrong account, or at least not being clear about which account's data it's showing. And those feel like outcomes that are important to avoid.

Adding that assert would kind of put this testing tactic out of reach, though, wouldn't it.